### DISCUSSION TITLE
Data vs Metadata

### DISCUSSION THREAD

**Discussion by Memphizzz** [jchristn/WatsonTcp#126] [Q&A]
Hi,

just a quick question regarding Metadata.
What's the advantage of using the Metadata dictionary vs the Data field?
I've only used the Data field in the past passing a serialized object to it. I assume metadata is also sent encrypted when using SSL.
Can you give me a typical example that would use Data and Metadata at the same time?

Thanks in advance!

P.S. Love your libraries! I'm also using WatsonWebServer :)

**Comment by jchristn** [comment]
Hi @Memphizzz thanks for the kind words!  

You can use data and metadata for really anything you want.  The main reasons I added metadata to the framing layer are 1) in case you want to stream data bidirectionally and don't want to have to include success/failure signals in the payload (i.e. the data will always be a file being transferred, or something along those lines, but you still need to tell the sender that it succeeded or failed) and 2) in case you want to send zero-byte messages (i.e. empty data) but still include some kind of signaling amongst nodes (i.e. in the metadata).

With SSL, everything sent is encrypted, including the full header (including metadata) and the data.

Cheers, Joel
